Makefile

```diff
@@ -167,6 +176,33 @@ else if (${PYTHON_VER//[^0-9]/} -ge 38)
 	@touch $(REQUIREMENT_DOT_FILE)
 endif
 
+.PHONY: virtual-env3.9
```

Comment:
```
.PHONY: virtual-env3.8
ifeq ($(PYTHON_VER),python3.8)
	virtual-env3.8: virtual-env3.x
endif

.PHONY: virtual-env3.9
ifeq ($(PYTHON_VER),python3.9)
	virtual-env3.9: virtual-env3.x
endif

.PHONY: virtual-env3.x
virtual-env3.x: $(BLD_DIR_ENV)/stamp
$(BLD_DIR_ENV)/stamp:
	@echo "--- Creating virtual environment at $(BLD_DIR_ENV) using $(PYTHON_VER)"
	@$(SYS_PYTHON) -m pip install --upgrade pip==22.2.2
	@$(SYS_PIP) install virtualenv==20.19.0 virtualenv-make-relocatable==0.0.1
	@if [[ "ppc64le" == $(PPC64LE) ]]; then \
	  $(SYS_PYTHON) -m venv $(BLD_DIR_ENV); \
	 fi
	@virtualenv -p $(PYTHON_VER) $(BLD_DIR_ENV)
	@echo "--- Virtual environment $(BLD_DIR_ENV) ready"
	@touch $@
	@echo '--- Installing PIP_MODULES in virtual-env'
	@if [[ "ppc64le" == $(PPC64LE) ]]; then \
	  echo '--- Installing $(REQUIREMENT_PPC64LE_FILE) into virtual-env via $(ENV_PIP)'; \
	  $(ENV_PIP) install -r $(REQUIREMENT_PPC64LE_FILE); \
	  echo '--- Finished $(REQUIREMENT_PPC64LE_FILE) into virtual-env'; \
	 else \
	  echo '--- Installing $(REQUIREMENT_FILE) into virtual-env via $(ENV_PIP)'; \
	  $(ENV_PIP) install -r $(REQUIREMENT_FILE); \
	  echo '--- Finished $(REQUIREMENT_FILE) into virtual-env'; \
         fi
	@$(ENV_PIP) install $(NAVOPTAPI_WHL)
	@echo '--- Finished $(NAVOPTAPI_WHL) into virtual-env'
	@touch $(REQUIREMENT_DOT_FILE)
```

@ranade1 Is it possible to use the duplicate code like that? instead of writing that again?

Reply:
It is possible to put it in one block. But there could be future chances where we may deviate.